support/sier-codec/src/lib.rs

```diff
@@ -48,13 +50,79 @@ impl Parser {
     pub fn struct_def(&self, name: &str) -> Option<&Arc<StructDef>> {
         self.structs.values().find(|s| s.type_name() == name)
     }
-}
 
-impl Default for Parser {
-    fn default() -> Self {
-        Parser {
-            structs: HashMap::new(),
+    fn transform_serde_value<'a>(
+        &mut self,
+        value: &SerdeValue,
+    ) -> Result<(Type, Value<'a>), Error<'a>> {
+        match value {
+            SerdeValue::Null => Ok((Type::Unit, Value::Unit)),
+            SerdeValue::Bool(_bool) => Ok((Type::Unit, Value::Unit)),
+            SerdeValue::Number(number) => {
+                if let Some(n) = number.as_u64() {
+                    Ok((Type::U64, Value::U64(n)))
+                } else {
+                    Ok((Type::Unit, Value::Unit))
+                }
+            }
+            SerdeValue::String(s) => Ok((Type::String, Value::String(s.to_string()))),
+            SerdeValue::Array(vec) => {
+                let mut list = vec![];
+                let mut arr_type = Type::Unit;
+                for arr_value in vec {
+                    let (val_type, val) = self.transform_serde_value(arr_value)?;
+                    arr_type = val_type;
+                    list.push(val);
+                }
+                Ok((Type::List(Box::new(arr_type)), Value::List(list)))
+            }
+            SerdeValue::Object(inner_json_obj) => {
+                let (_object, def) = self.transform_serde_obj(inner_json_obj)?;
+                Ok((
+                    Type::Struct(def),
+                    Value::Unit,
+                    //TODO(melatron): Figure how to handle the lifetime of object - Value::Struct(object),
```

Comment:
IIRC, our Value type has the lifetime parameter of the struct definitions we're working with. So, the 'a in the function definition should probably be the lifetime of the &self (shouldn't need to be mutable). I think changing `&mut self` to `&'a mut self` will let things compile (if they aren't already).